tests/test_bonded.py

```diff
@@ -117,6 +117,69 @@ def test_centroid_restraint_singularity(self):
                 )
 
 
+
+    def test_rmsd_restraint(self):
+        # test the RMSD force by generating random coordinates.
+
+        np.random.seed(2021)
+
+        box = np.eye(3) * 100
+
+        n_particles_a = 25
+        n_particles_b = 7
+
+        relative_tolerance_at_precision = {np.float32: 2e-5, np.float64: 1e-9}
+
+        for precision, rtol in relative_tolerance_at_precision.items():
+
+            for _ in range(100):
+                coords = self.get_random_coords(n_particles_a + n_particles_b, 3)
+
+                n = coords.shape[0]
+                n_mapped_atoms = 5
+
+                atom_idxs_a = np.arange(n_particles_a)
+                atom_idxs_b = np.arange(n_particles_b)
+
+                atom_map = np.stack([
+                    np.random.randint(0, n_particles_a, n_mapped_atoms, dtype=np.int32),
+                    np.random.randint(0, n_particles_b, n_mapped_atoms, dtype=np.int32) + n_particles_a
+                ], axis=1)
+
+                atom_map = atom_map.astype(np.int32)
+
+                params = np.array([], dtype=np.float64)
+                k = 1.35
```

Comment:
This is just an arbitrary value, because the functional form doesn't have a `b0` (it's implicitly zero in this case), we don't have to worry about the singularities present in the harmonic bond/center of mass restraints case.

Reply:
The precision here also caught my eye. In local tests, I confirmed that setting `k = 10000.0` for example doesn't change whether the test passes.